web/src/components/betaBar/BetaBar.test.tsx

```diff
@@ -0,0 +1,178 @@
+      expect((screen.getByLabelText("Feature request") as HTMLInputElement).value).toBe("random text");
+
+      fireEvent.click(screen.getByLabelText("close"));
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText));
+      expect((screen.getByLabelText("Feature request") as HTMLInputElement).value).toBe("");
+      expect(screen.queryByText(Config.betaBar.featureRequestModalInlineErrorText)).not.toBeInTheDocument();
+    });
+
+    it("submitting feature request renders submission successful modal", async () => {
+      useMockUserData({
+        formProgress: "COMPLETED",
+        profileData: generateProfileData({ hasExistingBusiness: false }),
+      });
+
+      render(<BetaBar />);
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText)).toBeInTheDocument();
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText));
+      fireEvent.change(screen.getByLabelText("Feature request"), { target: { value: "random text" } });
+      expect((screen.getByLabelText("Feature request") as HTMLInputElement).value).toBe("random text");
+
+      fireEvent.click(screen.getByText(Config.betaBar.featureRequestModalSubmitButtonText));
+
+      await waitFor(() => {
+        expect(screen.getByText(Config.betaBar.successfulSubmissionModalButtonText)).toBeInTheDocument();
+      });
+      expect(screen.queryByText(Config.betaBar.featureRequestModalInlineErrorText)).not.toBeInTheDocument();
+    });
+
+    it("does not submit feature request when input is empty", () => {
+      useMockUserData({
+        formProgress: "COMPLETED",
+        profileData: generateProfileData({ hasExistingBusiness: false }),
+      });
+      render(<BetaBar />);
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText)).toBeInTheDocument();
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText));
+      fireEvent.click(screen.getByText(Config.betaBar.featureRequestModalSubmitButtonText));
+      expect(screen.queryByText(Config.betaBar.successfulSubmissionModalButtonText)).not.toBeInTheDocument();
+      expect(screen.getByText(Config.betaBar.featureRequestModalInlineErrorText)).toBeInTheDocument();
+    });
+
+    it("clicking the submit another feedback button displays the beta modal", async () => {
+      render(<BetaBar />);
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText)).toBeInTheDocument();
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText));
```

Comment:
we're doing this series of 2 button clicks in nearly every one of these tests. I think it should be a helper function. Or perhaps a direct test of the Feedback component specifically so we don't have to open the BetaBar every single time?